### PR TITLE
Guns burst fire nerf

### PR DESCRIPTION
**Pull request by Waselon**
## About The Pull Request

Adds a base 0.15 extra_delay between burstfire, and finetunes most weapons burstfire to have the same DPS as autofire.

изображение

## Why It's Good For The Game

Fixes several guns burst fire having more DPS than autofire.

## Changelog
:cl:
balance: added 0.15 second base extra_delay, adjusted some weapons burstfire.
/:cl:

### CODE REVIEW

**Comment by srxtreme** [comment]
AR-18 extra delay should be further increased beyond 0.15, to 0.2 or even 0.3, due to having much higher DPS on autoburst than on semiauto, even with the burst delay at 0.15, 0.2 or 0.25.

**Comment by FIXTHERADBOXPLEASE** [comment]
It was always kind of weird that burst-mode shoots far faster than full-auto, when in real life burst-modes are designed to trade speed for accuracy.
I mean, TGMC burst mode even auto-tracks the target for you if you click the sprite. It's both faster AND more accurate. Full-auto is a pointless noobtrap as is.

**Comment by srxtreme** [comment]
Salt PR?

**Comment by Waselon** [comment]
> Salt PR?

Yes.

**Comment by collinlunn** [comment]
IMO full-auto is currently obsolete on some guns, as burstfire has both far better DPS _and_ more upfront damage, and I agree that is a bad thing. Why have a feature on the guns that no one would want to use?

However, there may be an alternative way to resolve this, depending on opinions from the balance maintainers.
It may also be possible to:
1. Remove the "obsolete" full-auto from relevant guns.
2. Add other penalties to burst-fire to compensate
3. For the currently presented solution of reducing burst DPS, it's not 100% clear to me what % reduction in DPS the changes in delay will induce. Maybe it's possible to have the gun automatically calculate its delay with an equation involving the full-auto `fire_delay` and `burst_amount` to give more clarity to how burst DPS relates to full-auto DPS?

**Comment by Lumipharon** [comment]
Burst fire is not intended to be slower than auto fire. It has explicitly higher dps, at the cost of worse scatter. Those scatter penalties were severely reduced on most guns though, which I thought wasn't a great idea as it removed most of the drawback.

Tgmc burstfire is not rl burstfire.

**Comment by Strayed-Raven** [comment]
This is one thirds not knowing that its intended or i suppose what is. one third ignoring scatter balancing, which is also taking into consideration now I believe the scatter increase as you fire mechanic? and the other third is just a bit of salt imo. Otherwise what is the point of attachments? Also its coping about akimbo, again. Make up your mind people.

**Comment by Waselon** [comment]
> Burst fire is not intended to be slower than auto fire. It has explicitly higher dps, at the cost of worse scatter. Those scatter penalties were severely reduced on most guns though, which I thought wasn't a great idea as it removed most of the drawback.
> 
> Tgmc burstfire is not rl burstfire.

Please explain why should I be able to switch fire modes for free +50% DPS?

**Comment by RipGrayson** [comment]
I give it 3 days for someone to increase scatter to pre-nerf values like Lumi suggested, or this PR gets merged. Either way something has to be done.

**Comment by FIXTHERADBOXPLEASE** [comment]
Apparently burst fire wasn't balanced around scatter, but higher initial damage for lower sustained DPS. Just forwarding what I heard on the balance channel.

**Comment by Lumipharon** [comment]
> 

Burst dps hasn't changed, atkeast for 2 years and likely far longer. The more accurate question is why have auto fire on guns designed for burstfire like t18 and t11. As for the dps question between t18 and hsg: t18 has 36 bullets in a mag and 5 ap. Hsg has like 40 ap, so is far better against armoured targets, has IFF, has a large mag for sustained fire, and far higher single shot damage and better damage falloff, so is much better over longer distances.

**Comment by FIXTHERADBOXPLEASE** [comment]
> In what world would you ever want to drop your overall dps to apply like 40 damage 0.3 seconds faster?
The amount of time you have to actually shoot a Xenomorph is very short. High damage in a faster period of time is what kills them, not sustained fire, unless you're in a chase.

Although yeah burstfire is patently superior codewise, and it's questionable why some guns even have autofire. All it really needs is some fine-tuned numbers tweaks. The t-18 in particular is definitely too strong.

**Comment by Waselon** [comment]
изображение

**Comment by Barnet2** [comment]
> изображение

If this chart is to be true, you legit gave AR-12 and AR-18 the same DPS. 
What.

**Comment by Waselon** [comment]
> If this chart is to be true, you legit gave AR-12 and AR-18 the same DPS. What.

Ok and?

**Comment by FIXTHERADBOXPLEASE** [comment]
The ar-18 was always just a more lighter, more mobile version of the ar-12.

**Comment by Lumipharon** [comment]
> изображение

Aye, and if we use penetration and fall off, we get some more useful figures.
Against an ancient warrior with no warding at 5 tiles:
T18 is 51.25 dps auto, 74.62 burst.
T12 is 56.25 auto, 61.2 burst
Tx11 is 45.5 auto, 75.64 burst
T19 is 50 auto, 68.42 burst
Hsg is 119 auto, 60.98 burst (it gets like double accuracy though)

This doesn't take into account scatter, accuracy or sunder however.

**Comment by KerbalCitizen** [comment]
Don't remove full auto on guns imo. Some people like it for niche purposes like shooting huggers and only using one bullet. If you personally don't like full auto because you prefer higher dps, then just use burst. Don't force other people to change to your playstyle.

**Comment by Zar000** [comment]
Maybe full auto should actually have higher DPS to keep it useful and burst should have insane scatter that expands for every bullet after the first one.

Example
5 round burst
Round 
1 100% acc
2. 90% acc
3. 80% acc
ETC

**Comment by SilencedMP5A5** [comment]
local man has no idea how guns with a burst fire mode work

also the ENTIRE point of the T18 and the TX11 is to be excellent burst rifles

salt PR

**Comment by RipGrayson** [comment]
This PR is still being discussed, I'm not merging anything here, just to clarify my earlier comment.

**Comment by Strayed-Raven** [comment]
With the sunder PR merged and other Xeno QoL PRs. This one is just a snowball PR that should be denied. Tweak xenos more. Let skill issues lie where skill issues lie.

**Comment by srxtreme** [comment]
bold claim coming from someone who has a +50% dps switch.
